File: 3042-count-prefix-and-suffix-pairs-I/main.cpp
class Solution {
public:
    int countPrefixSuffixPairs(vector<string>& words) {
        size_t n = words.size();
        int count = 0;

        for (size_t i = 0; i < n; i++) {
            for (size_t j = i + 1; j < n; j++) {
                if (words[j].starts_with(words[i]) && words[j].ends_with(words[i])) {
                    count++;
                }
            }
        }

        return count;
        
    }
};

File: 3042-count-prefix-and-suffix-pairs-I/main.go
func countPrefixSuffixPairs(words []string) int {

    count := 0
    for i := range words {
        for j := i + 1; j < len(words); j++ {
            if strings.HasPrefix(words[j], words[i]) && strings.HasSuffix(words[j], words[i]) {
                count++
            }
        }
    }

    return count
}
